lighthouse-core/audits/lazy-third-party.js

```diff
@@ -0,0 +1,183 @@
+
+const UIStrings = {
+  /** Title of a diagnostic audit that provides details about the third-party code on a web page that can be lazy loaded. This descriptive title is shown to users when no resources have lazy loading alternatives available. */
+  title: 'Lazy load third-party resources',
+  /** Title of a diagnostic audit that provides details about the third-party code on a web page that can be lazy loaded. This descriptive title is shown to users when one or more third-party resources have available lazy loading alternatives. */
+  failureTitle: 'Some third-party resources can be lazy loaded',
+  /** Description of a Lighthouse audit that identifies the third party code on the page that can be lazy loaded. This is displayed after a user expands the section to see more. No character length limits. 'Learn More' becomes link text to additional documentation. */
+  description: 'Some third-party resources can be fetched after the page loads. ' +
+    'These third-party resources are used by embedded elements which can be replaced by a facade ' +
+    'until the user needs to use them. [Learn more]().',
+  /** Summary text for the result of a Lighthouse audit that identifies the third party code on a web page that can be lazy loaded. This text summarizes the number of lazy loading facades that can be used on the page. */
+  displayValue: `{itemCount, plural,
+  =1 {# facade alternative available}
+  other {# facade alternatives available}
+  }`,
+  /** Label for a table column that displays the name of a lazy loading facade alternative for a third party resource. */
+  columnFacade: 'Facade Alternative',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+/** @typedef {import("third-party-web").IEntity} ThirdPartyEntity */
+/** @typedef {import("third-party-web").IFacade} ThirdPartyFacade */
+
+/** @typedef {{mainThreadTime: number, transferSize: number, blockingTime: number}} Summary */
+/** @typedef {{summary: Summary, urlSummaries: Map<string, Summary>}} FacadeSummary */
+
+class LazyThirdParty extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'lazy-third-party',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['traces', 'devtoolsLogs', 'URL'],
+    };
+  }
+
+  /**
+   *
+   * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
+   * @param {Array<LH.Artifacts.TaskNode>} mainThreadTasks
+   * @param {number} cpuMultiplier
+   * @param {ThirdPartyEntity | undefined} mainEntity
+   * @return {Map<ThirdPartyFacade, FacadeSummary>}
+   */
+  static getSummaries(networkRecords, mainThreadTasks, cpuMultiplier, mainEntity) {
+    /** @type {Map<ThirdPartyFacade, FacadeSummary>} */
+    const summaries = new Map();
+    const defaultStat = {mainThreadTime: 0, blockingTime: 0, transferSize: 0};
```

Comment:
stat -> summary, since these are `FacadeSummary`s and the resultant map is `summaries` ?

Reply:
(oh, you'll end up reusing this when that other pr lands, nvm)